### ISSUE TITLE
One Piece TVDB4 Specials misbehaving

### ISSUE THREAD

**Issue by TrafalgaT** [ZeroQI/Absolute-Series-Scanner#288]
I noticed recently that all specials past special 30 (S31 - One Piece: Episode of Sabo) get meta matched to the episode 31 instead of the special

[One Piece [tvdb4-81797].scanner.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4850671/One.Piece.tvdb4-81797.scanner.log)
[One Piece [tvdb4-81797].filelist.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4850672/One.Piece.tvdb4-81797.filelist.log)

**Comment by sven-7** [comment]
I made several updates in ScudLee where One Piece specials were off or messed up. Not sure if this will ultimately close your issue or not, but brought light to that at the very least. Keep an eye out for that commit.

**Comment by ZeroQI** [comment]
Please confirm, it should be resolved

**Comment by TrafalgaT** [comment]
Hi, so...no i tried refreshing the meta for s00 still as it used to be, i removed the folder and moved it elsewhere, readded it and its still being matched wrong

**Comment by ZeroQI** [comment]
I will need the scanner logs, and one special number with issues as i need to check the mapping file, anidb and thetvdb for it...

**Comment by TrafalgaT** [comment]
[One Piece [tvdb4-81797].filelist.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4877876/One.Piece.tvdb4-81797.filelist.log)
[One Piece [tvdb4-81797].scanner.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4877877/One.Piece.tvdb4-81797.scanner.log)


any special past 30 so for example special 31, or the recent one 39 stampede (they all get matched as episodes instead of specials)

**Comment by ZeroQI** [comment]
I do not see any specials in either logs

```
Library: 'no valid X-Plex-Token.id', root: '/share/CACHEDEV2_DATA/TV Anime', path: 'One Piece [tvdb4-81797]/S00 - Specials', files: '0', dirs: '2'
Plex scan start: 2020-07-06 09:59:32,849064
plexignore_files: '[]', plexignore_dirs: '[]'
```
```
[folder] One Piece [tvdb4-81797]/S00 - Specials/[0.1] Arc Movies
[folder] One Piece [tvdb4-81797]/S00 - Specials/[0.2] Arc Special Episodes
[no files detected] Grouping folder skip as >1 folder [return]
```
No episode 39 can be found in either logs...

**Comment by TrafalgaT** [comment]
yes, i saw that also, is there a special reason it keeps skipping the folder ?

**Comment by ZeroQI** [comment]
How can the metadata be wrong when you say "it keeps missing the s00 folder now "
The second phrase in the readme states: 

> If a file is not showing in plex or showing at the wrong season and/or episode number, or is not passing through the forced id, then it is a scanner issue. Anything else is metadata related and agent specific.

I don't think you should have a folder per episode, especially not for specials, especially not in a grouping folder, not sure if tvdb4 mode doesn't add itself complexity there...

The scanner will not re-scan if the folder content didn't change... you might have to do the Plex Dance...

**Comment by TrafalgaT** [comment]
Ok I think you are not understanding what I am saying.

Initally I added the Series (with tvdb4-#) as a folder, grouped by specials and seasons, subgrouped by arcs. All of this worked flawlessly, until recently when i noticed that the specials were not being matched properly. Everything before special episode 31 got matched properly, everything after 31 is not getting matched. This is not a new process. I realised this when i added the latest special Stampede, which wrongly matched to epsiode 38 of the series. so to answer your intial accusation. the metadata was correctly matched for everything else, all the seasons are fine. This issue just concerns the specials folder and all episodes titled 30+ consecutively.

**Comment by ZeroQI** [comment]
Thanks for the precision.
The files are added to Plex but I cannot see them in scanner logs (which makes things tricky), but **if you tell me the specials 31+ ends up in Plex with the exact same episode number and season 0**, then we can skip the scanner logs, as you assign manually yourself the season number so the scanner is transparent on the season and episode number assignment

If that is the situation, then it is a metadata issue and we need to check the mapping so the agent logs would help to know if it uses the latest mapping files or a cached version
It would also show which source provided the meta, and the dual way mapping used between ANiDB and TheTVDB

TheTVDB s00e31 https://thetvdb.com/series/one-piece/episodes/5279341
AniDB S00e21 https://anidb.net/episode/170966
```
  <anime anidbid="69" tvdbid="81797" defaulttvdbseason="a" episodeoffset="" tmdbid="" imdbid="">
    <name>One Piece</name>
    <mapping-list>
      <mapping anidbseason="0" tvdbseason="0">;1-28;2-3;3-9;4-10;5-14;6-0;7-0;8-0;9-0;10-0;11-0;12-0;13-0;14-0;15-0;16-32;17-23;18-24;19-29;20-30;21-31;22-0;23-33;24-35;25-0;26-37;27-38;</mapping>
    </mapping-list>
  </anime>
```
The mapping file is correct and your episode number should be s00e31 in Plex using TVDB4 mode
I need the agent logs to SEE the metadata source used, and which provided wrong info and if the agent has loaded the right mapping from ScudLee's XML mapping files
